test/testutils.jl

```diff
@@ -0,0 +1,330 @@
+            # Test having only missing data
+            c = DataFrame(
+                :sin => fill(missing, 10),
+                :cos => fill(missing, 10),
+            )
+            if tester.imp == DropObs
+                @test impute(c, tester.imp(; tester.kwargs...)) == DataFrame()
+            elseif tester.imp == DropVars
+                # https://github.com/JuliaData/Tables.jl/issues/117
+                @test_broken impute(c, tester.imp(; tester.kwargs...)) == DataFrame()
+            else
+                @test isequal(impute(c, tester.imp(; tester.kwargs...)), c)
+            end
+        end
+
+        @testset "Too many missing values" begin
+            # Test Context error condition
+            c = DataFrame(
+                :sin => fill(missing, 10),
+                :cos => fill(missing, 10),
+            )
+            settings = deepcopy(tester.kwargs)
+            settings[:context] = Context(; limit=0.1)
+            if tester.imp == DropVars
+                # https://github.com/JuliaData/Tables.jl/issues/117
+                @test_throws MethodError impute(c, tester.imp(; settings...))
+                @test_throws MethodError tester.f(c; settings...)
+            else
+                @test_throws ImputeError impute(c, tester.imp(; settings...))
+                @test_throws ImputeError tester.f(c; settings...)
+            end
+        end
+    end
+end
+
+function test_axisarray(tester::ImputorTester)
+    @testset "AxisArray" begin
+        a = allowmissing(1.0:1.0:20.0)
+        a[[2, 3, 7]] .= missing
+        m = collect(reshape(a, 5, 4))
+        aa = AxisArray(
+            deepcopy(m),
+            Axis{:time}(DateTime(2017, 6, 5, 5):Hour(1):DateTime(2017, 6, 5, 9)),
+            Axis{:id}(1:4)
+        )
+        result = impute(aa, tester.imp(; tester.kwargs...))
+
+        @testset "Base" begin
+            # Test that we have fewer missing values
+            @test count(ismissing, result) < count(ismissing, aa)
+            @test isa(result, AxisArray)
+            @test eltype(result) <: eltype(aa)
+
+            # Test that functional form behaves the same way
+            @test result == tester.f(aa; tester.kwargs...)
+        end
+
+        @testset "In-place" begin
+            # Test that the in-place function return the new results and logs whether it
+            # successfully did it in-place
+            aa2 = deepcopy(aa)
+            aa2_ = tester.f!(aa2; tester.kwargs...)
+            @test aa2_ == result
+            if aa2 != result
+                @warn "$(tester.f!) did not mutate input data of type AxisArray"
+            end
+        end
+    end
+end
+
+function test_columntable(tester::ImputorTester)
+    @testset "Column Table" begin
+        table = DataFrame(
+            :sin => allowmissing(sin.(1.0:1.0:20.0)),
+            :cos => allowmissing(sin.(1.0:1.0:20.0)),
+        )
+
+        table.sin[[2, 3, 7, 12, 19]] .= missing
+        coltab = Tables.columntable(table)
```

Comment:
I can recall if this will convert a DataFrame to a "canonical" column table. (I.e a Named Tuple of Vectors)
Or if it will be a no-op.

Either way something is wrong.
Either should not have created the DataFrame to begin with,or should not have done this line.

Though it might be nice to test both.

Reply:
This was a holdover from the old tests. I should just construct a named tuple of arrays instead of a dataframe here.